openeo/extra/artifacts/exceptions.py

```diff
@@ -0,0 +1,61 @@
+class ArtifactsException(Exception):
+    """
+    Family of exceptions related to artifacts
+    """
+
+
+class NoAdvertisedProviders(ArtifactsException):
+    """
+    The providers list is empty. This OpenEO backend does not seem to support artifacts or at least not advertise its
+    capabilities.
+    """
+
+    def __str__(self):
+        return """
+The OpenEO backend used does not advertise providers for managing artifacts.
+""".lstrip()
+
+
+class UnsupportedArtifactsType(ArtifactsException):
+    """
+    The artifacts type is not supported. This should only occur if you specify config manually and the artifact type
+    is not supported by the backend.
+
+    :parameter type_id: The type_id that is not supported but which was attempted to be used
+    """
+
+    def __init__(self, type_id: str):
+        self.type_id = type_id
+
+    def __str__(self):
+        return f"The OpenEO backend does not support {self.type_id}"
```

Comment:
I'm not sure about the pattern to override `__str__` in exception subclasses. Often, exceptions are `repr`ed instead of `str`ed, and then you won't get that error message.

I think it's more robust and simpler, to do everything in `__init__`, e.g.

```python
class UnsupportedArtifactsType(ArtifactsException):
    def __init__(self, type_id: str):
        super().__init__(f"The OpenEO backend does not support {type_id}")
```

(if you would still want to be able to recover the `type_id` from the exception, you can still attach it as attribute in `__init__` of course, but I left that out for clarity)